craft_application/models/project.py

```diff
@@ -38,6 +40,13 @@
     VersionStr,
 )
 
+CURRENT_DEVEL_BASE = craft_providers.bases.ubuntu.BuilddBaseAlias.NOBLE
+DEVEL_BASE_WARNING = (
```

Comment:
we can put this warning in craft-application but personally I think it's secondary, and separate from the validation that we want to do

Reply:
What do you think about putting it [here](https://github.com/canonical/craft-application/blob/c36bb63950df3ef64dc7e8362800f44d7093e327/craft_application/services/provider.py#L118)? 

In that location, users will only see the warning when launching a `devel` instance, not everytime the project is parsed.